app/main/views/drafts.py

```diff
@@ -264,27 +268,32 @@ def create_new_draft_service(framework_slug):
     """
     updater_json = validate_and_return_updater_request()
     draft_json = validate_and_return_draft_request()
-    json_has_required_keys(draft_json, ['lot', 'supplierId'])
 
-    framework = Framework.query.filter(
-        Framework.slug == framework_slug
-    ).first()
+    if framework_slug:
+        current_app.logger.warning("Deprecated /draft-services/<framework_slug> route")
+        draft_json['frameworkSlug'] = framework_slug
 
-    if framework.status != 'open':
-        abort(400, "'{}' is not open for submissions".format(framework_slug))
+    framework, lot, supplier = validate_and_return_related_objects(draft_json)
+    if lot.one_service_limit:
+        lot_service = DraftService.query.filter(DraftService.supplier == supplier, DraftService.lot == lot).first()
+        if lot_service:
+            abort(400, "'{}' service already exists for supplier '{}'".format(lot.slug, supplier.supplier_id))
 
-    supplier_id = draft_json['supplierId']
-    errs = get_draft_validation_errors(draft_json, slug=framework_slug)
-    if errs:
-        return jsonify(errors=errs), 400
+    if framework.status != 'open':
```

Comment:
I might put this check before the `one_service_limit` check - seems like if the framework isn't open then we shouldn't even start looking into the services and lots for it.

But it doesn't really matter...